### PR TITLE
MFA feature: Enable google authenticator

### PR DESCRIPTION
**Pull request by abhilashraju**
Enabling multifactor authentication for BMC. This feature enables google authenticator using TOTP method.

Change-Id: I053095763c65963ff865b487ab08f05039d2fc3a

### CODE REVIEW

**Comment by geissonator** [comment]
add to approvelist

File: users.cpp [hunk 1]
```diff
@@ -194,5 +198,79 @@ bool Users::userPasswordExpired(void) const
...
+{
+    if (!std::filesystem::exists(authAppPath))
+    {
+        lg2::error("No authenticator app found at {PATH}", "PATH", authAppPath);
+        return;
+    }
+    std::string path = std::format(secretKeyTempPath, userName);
+    executeCmd(authAppPath.data(), "-s", path.c_str(), "-u", "-W", "-Q", "NONE",
+               "-t", "-f", "-D", "-C");
+    // isSecretKeySetup(true);
```
Comment (sunharis) [inline users.cpp:211]: commented code can be removed

File: users.cpp [hunk 2]
```diff
@@ -194,5 +198,79 @@ bool Users::userPasswordExpired(void) const
...
+}
+struct MFABypassHandlers
+{
+    MFABypassType type;
+    std::function<bool(Users&)> handler;
+};
+static bool emptyfunc(Users& /*unused*/)
```
Comment (sunharis) [inline users.cpp:218]: What is the use of this emptyfunc ?

File: users.cpp [hunk 3]
```diff
@@ -194,5 +198,79 @@ bool Users::userPasswordExpired(void) const
...
 
+void Users::createSecretKey()
+{
+    if (!std::filesystem::exists(authAppPath))
+    {
+        lg2::error("No authenticator app found at {PATH}", "PATH", authAppPath);
+        return;
+    }
+    std::string path = std::format(secretKeyTempPath, userName);
+    executeCmd(authAppPath.data(), "-s", path.c_str(), "-u", "-W", "-Q", "NONE",
```
Comment (sunharis) [inline users.cpp:209]: This created temp file. But where is the permanent secret key file is created ? There should be a rename of this file to the permanent file right ?

**Pull request closed without merging**